### PR TITLE
Show diff in xunit

### PR DESCRIPTION
**Pull request by mlucool**
### Description of the Change
XUnit reporter should show diffs as output

### Why should this be in core?
XUnit reporter was already in core

### Benefits
You can tell actual vs. expected which is very helpful and the default in the default reporter.

### Possible Drawbacks
People wrote parsers.

### Applicable issues
This is an enhancement.

### CODE REVIEW

**Comment by mlucool** [comment]
@outsideris All requested changes have been fixed. Please let me know if anything else is required.

**Comment by mlucool** [comment]
@outsideris any thoughts?

**Comment by outsideris** [comment]
@mlucool I will review it this weekend.

**Comment by mlucool** [comment]
@outsideris did you get a chance to look at this?

**Comment by mlucool** [comment]
@boneskull can you take a look? It seems @outsideris is caught up IRL :)

**Comment by plroebuck** [comment]
@mlucool, you never responded...

Does this still function as desired if `Base.inlineDiffs = true;` XML output-wise?

**Comment by mlucool** [comment]
@plroebuck this will not work well for inlineDiffs as there is no color in XML. I verified that inline diffs don't work in anywhere without colors (via `xterm -cm`), so this maintains and extends status quo for that feature. What would you like to do here?

**Comment by plroebuck** [comment]
@mlucool, your change brings this question up -- please add your opinion to #3537, but it's not so much directed at you...

**Comment by plroebuck** [comment]
@mlucool, whether it "looks good" is beside the point; is it "functional" (i.e., can you discern differences without color) using inline diffs or not? If so, that's fine. If not... what's plan b?

Up until this change, `--color`/`--no-color` was moot.

**Comment by mlucool** [comment]
@plroebuck I apologize, I should have been more clear above. Let me try again. If you open up a terminal which does not support colors (e.g. via `xterm -cm`) then run the default reporter, `spec`, you also cannot understand the diff correctly. I am happy to update this commit to throw or warn in this case, which is likely the best thing if treated in isolation, but I was suggesting above that we do something a little more globally than just for xunit.

**Comment by plroebuck** [comment]
@mlucool my apologies for having missed this reply here. Given an incredibly short time frame, would you agree the PR is adequate for normal usage, and color can be resolved in bugfix release if required?

**Comment by mlucool** [comment]
@plroebuck totally agree because the default is not the inline version
